psa-crypto/src/types/key.rs

```diff
@@ -259,15 +259,23 @@ impl Attributes {
                     false
                 }
             }
-            Type::EccKeyPair { .. } | Type::EccPublicKey { .. } => {
-                if let Algorithm::AsymmetricSignature(sign_alg) = alg {
-                    sign_alg.is_ecc_alg()
-                } else {
-                    false
-                }
-            }
+            Type::EccKeyPair { .. } | Type::EccPublicKey { .. } => match alg {
+                Algorithm::KeyAgreement(KeyAgreement::Raw(RawKeyAgreement::Ecdh))
+                | Algorithm::KeyAgreement(KeyAgreement::WithKeyDerivation {
+                    ka_alg: RawKeyAgreement::Ecdh,
+                    ..
+                }) => true,
+                Algorithm::AsymmetricSignature(sign_alg) => sign_alg.is_ecc_alg(),
+                Algorithm::AsymmetricEncryption(_) => true,
```

Comment:
Hmm I think I told you something wrong, there don't seem to be any [asymmetric encryption algorithms](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_algorithm.html#asymmetricencryption-algorithm) for ECC keys.